library/std/src/sys/net/connection/socket/unix.rs

```diff
@@ -485,14 +485,18 @@ impl Socket {
 
     // bionic libc makes no use of this flag
     #[cfg(target_os = "linux")]
-    pub fn set_deferaccept(&self, accept: u32) -> io::Result<()> {
-        setsockopt(self, libc::IPPROTO_TCP, libc::TCP_DEFER_ACCEPT, accept as c_int)
+    pub fn set_deferaccept(&self, accept: Duration) -> io::Result<()> {
+        let val = accept.as_secs();
+        if val > i32::MAX as _ {
+            return Err(io::Error::from_raw_os_error(libc::EINVAL));
+        }
```

Comment:
I'd use `try_into()` to make the conversion safer. Also, it might be good to use `io::const_error!` instead of a raw error since this is not an error returned from the OS (and thus not explained in any man-page). That would also allow attaching a message describing the error.